code/modules/shuttle/supply.dm

```diff
@@ -115,13 +159,37 @@ GLOBAL_LIST_INIT(blacklisted_cargo_types, typecacheof(list(
 		SSshuttle.shoppinglist -= SO
 		SSshuttle.orderhistory += SO
 
-		SO.generate(pick_n_take(empty_turfs))
+		if(SO.pack.small_item) //small_item means it gets piled in the miscbox
+			if(SO.paying_account)
+				if(!miscboxes.len || !miscboxes[D.account_holder]) //if there's no miscbox for this person
+					miscboxes[D.account_holder] = new /obj/structure/closet/crate/secure/owned(pick_n_take(empty_turfs), SO.paying_account)
+					miscboxes[D.account_holder].name = "small items crate - purchased by [D.account_holder]"
+				for (var/item in SO.pack.contains)
+					new item(miscboxes[D.account_holder])
+				misc_order_num[D.account_holder] = "[misc_order_num[D.account_holder]]#[SO.id]  "
+			else //No private payment, so we just stuff it all into a generic crate
+				if(!miscboxes.len || !miscboxes["Cargo"])
+					miscboxes["Cargo"] = new /obj/structure/closet/crate/secure(pick_n_take(empty_turfs))
+					miscboxes["Cargo"].name = "small items crate"
+					miscboxes["Cargo"].req_access = list()
+				for (var/item in SO.pack.contains)
+					new item(miscboxes["Cargo"])
+				if(SO.pack.access)
+					miscboxes["Cargo"].req_access += SO.pack.access
+				misc_order_num["Cargo"] = "[misc_order_num["Cargo"]]#[SO.id]  "
+		else
+			var/obj/structure/closet/crate/crate = SO.generate(pick_n_take(empty_turfs))
+			generateManifest(crate, "", "#[SO.id]", SO.pack.name)
```

Comment:
And you call the same function here, without any grouped or not argument passed in.